app/models/form526_submission.rb

```diff
@@ -308,42 +310,29 @@ def perform_ancillary_jobs(first_name)
   def workflow_complete_handler(_status, options)
     submission = Form526Submission.find(options['submission_id'])
     if submission.jobs_succeeded?
-      submission.send_form526_confirmation_email(options['first_name'])
+      Form526ConfirmationEmailJob.perform_async(personalization_parameters)
       submission.workflow_complete = true
       submission.save
     else
-      submission.send_form526_submission_failed_email(options['first_name'])
+      Form526SubmissionFailedEmailJob.perform_async(personalization_parameters)
     end
   end
 
-  def send_form526_confirmation_email(first_name)
-    email_address = form['form526']['form526']['veteran']['emailAddress']
-    personalization_parameters = {
-      'email' => email_address,
-      'submitted_claim_id' => submitted_claim_id,
-      'date_submitted' => created_at.strftime('%B %-d, %Y %-l:%M %P %Z').sub(/([ap])m/, '\1.m.'),
-      'first_name' => first_name
-    }
-    Form526ConfirmationEmailJob.perform_async(personalization_parameters)
+  def bdd?
+    form.dig('form526', 'form526', 'bddQualified') || false
   end
 
-  def send_form526_submission_failed_email(first_name)
-    email_address = form['form526']['form526']['veteran']['emailAddress']
-    personalization_parameters = {
-      'email' => email_address,
+  private
+
+  def personalization_parameters
+    {
+      'email' => form['form526']['form526']['veteran']['emailAddress'],
       'submitted_claim_id' => submitted_claim_id,
       'date_submitted' => created_at.strftime('%B %-d, %Y %-l:%M %P %Z').sub(/([ap])m/, '\1.m.'),
-      'first_name' => first_name
+      'first_name' => get_first_name
```

Comment:
Call `get_first_name` here when it is needed. We don't ever set `first_name` to anything other than the result of calling `get_first_name`.